.github/workflows/build-container.yml

```diff
@@ -22,63 +99,140 @@ jobs:
       GPG_KEY_PASSWORD: ${{ secrets.GPG_KEY_PASSWORD }}
       DOCKER_HUB_ACCESS_TOKEN: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
       DOCKER_HUB_USER: ${{ secrets.DOCKER_HUB_USER }}
+      LOCAL_REPO: /maven
     steps:
       - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
 
-      - name: Install deps
+      - name: Work around git permission issue
         run: |
-          apk add libxml2-utils docker
+          dname=$(echo ${{github.repository}} | cut -d'/' -f2)
+          git config --global --add safe.directory /__w/$dname/$dname
+        shell: sh
 
-      - name: Packer Version
+      - name: Build repository
+        id: build
         run: |
-          packer version
+          # We have just checked out the top HEAD,
+          # so save the build script separately
+          # to use for building everything
+          cp for-ci-build-image/maven-build.sh .
+
+          # parse commits as a list
+          commits=( ${{ needs.parse-inputs.outputs.commits }} )
+          for commit in "${commits[@]}"; do
+            # reset the repository to clear any changes
+            # so that we can checkout again
+            # we're already in the in-repo path, so no need to specify it
+            git reset --hard $commit
+
+            # build the deps
+            cd for-ci-build-image
+            ../maven-build.sh 
+            cd ..
```

Comment:
todo: Assuming the above was made to work, this is changing into a directory that potentially has files that doesn't match what is in the script.